docs/worldgen/biomemodifier.md

```diff
@@ -0,0 +1,648 @@
+
+
+- Avoid adding the same placed feature with more than one Biome Modifier, as this can cause feature cycle violations.
+
+### The available values for the Decoration steps
+
+The `step` field in many of these JSONs are referring to GenerationStep.Decoration enum. This enum has the steps listed out in this order which is the same order that the game uses for generating during worldgen:
+
+    raw_generation
+    lakes
+    local_modifications
+    underground_structures
+    surface_structures
+    strongholds
+    underground_ores
+    underground_decoration
+    fluid_springs
+    vegetal_decoration
+    top_layer_modification
+
+
+## Datagenning Biome Modifiers:
+
+A `BiomeModifier` can be with [data generation][datagen] by passing a `RegistrySetBuilder` to `DatapackBuiltinEntriesProvider`. Biome Modifiers are located within `data/<modid>/neoforge/biome_modifier/<path>.json` All Biome Modifiers contain a `type` key that references the id of the `MapCodec` used for the Biome Modifier. All other settings provided by the Biome Modifier are added as additional keys on the root object.
+
+```java
+// Define keys for datapack registry objects
+
+public static final ResourceKey<BiomeModifier> EXAMPLE_MODIFIER =
+    ResourceKey.create(
+        NeoForgeRegistries.Keys.BIOME_MODIFIERS, // The registry this key is for
+        ResourceLocation.fromNamespaceAndPath(MOD_ID, "example_modifier") // The registry name
+    );
+
+// For some RegistrySetBuilder BUILDER
+//   being passed to DatapackBuiltinEntriesProvider
+//   in a listener for GatherDataEvent
+BUILDER.add(NeoForgeRegistries.Keys.BIOME_MODIFIERS, bootstrap -> {
+    // Lookup any necessary registries
+    // Static registries only need to be looked up if you need to grab the tag data
+    HolderGetter<Biome> biomes = bootstrap.lookup(Registries.BIOME);
+
+    // Register the Biome Modifiers
+
+    bootstrap.register(EXAMPLE_MODIFIER,
+        new ExampleBiomeModifier(
+            biomes.getOrThrow(Tags.Biomes.IS_OVERWORLD),
+            20
+        )
+    );
+})
+```
+
+```json5
+// In data/examplemod/neoforge/biome_modifier/example_modifier.json
+{
+    // The registy key of the MapCodec for the modifier
+    "type": "examplemod:example_biome_modifier",
+    // All additional settings are applied to the root object
+    "biomes": "#c:is_overworld",
+    "value": 20
+}
+```
+
+### Datagenning Existing Biome Modifiers
+
+NeoForge provides some basic Biome Modifiers for common usecases. The datagen implementation will be shown. All Biome Modifiers can be found in `BiomeModifiers`.
```

Comment:
I feel like having this separate and duplicated in description is a bit redundant.

Reply:
I did not want to have the datagen code mixed with the json code as it'll bog down and overwhelm players and pack makers. Hence why I split it out.

Reply:
Understandable, though I think we generally have them close together in the docs to avoid the duplicate information. It may be worthwhile then to use docusaurus tabs and have the java representation for generation be selected from.